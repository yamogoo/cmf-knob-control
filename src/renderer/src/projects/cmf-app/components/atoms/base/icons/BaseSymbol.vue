<template lang="pug">
component(
  :is="component"
  v-if="show"
  :class="[`${APP_PREFIX}-icon`]"
)
</template>

<script setup lang="ts">
import { APP_PREFIX } from '@app/config'
import { defineAsyncComponent } from 'vue'

export interface Props {
  path: string
  show?: boolean
}

const props = withDefaults(defineProps<Props>(), {
  format: 'vue',
  show: true
})

const component = defineAsyncComponent(() => import(props.path))
</script>

<script lang="ts">
export * from './icons'
</script>

<style lang="scss">
.#{$APP_PREFIX} {
  &-icon {
    fill: inherit;

    path: {
      fill: inherit;
    }
  }
}
</style>
