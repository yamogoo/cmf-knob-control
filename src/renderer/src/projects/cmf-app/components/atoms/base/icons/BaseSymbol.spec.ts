import { mount } from '@vue/test-utils'
import { describe, expect, test, vi } from 'vitest'

import BaseSymbol from './BaseSymbol.vue'

describe('BaseSymbol', () => {
  test.each(['./symbols/AttentionSymbol.vue', './symbols/AutoSymbol.vue'])(
    'should render svg DOM element (path: %s)',
    async (path) => {
      const component = mount(BaseSymbol, {
        props: {
          path
        }
      })

      await vi.dynamicImportSettled()

      const svg = component.find('svg')
      const isSVGExis = svg.exists()

      expect(isSVGExis).toBe(true)
      expect(isSVGExis).toMatchSnapshot()
    }
  )

  test.each([''])('should render ', async () => {
    const component = mount(BaseSymbol, {
      props: {
        path: './symbols/AttentionSymbol.vue'
      }
    })

    await vi.dynamicImportSettled()

    const svg = component.find('svg')
    const isSVGExis = svg.exists()

    expect(isSVGExis).toBe(true)
    expect(isSVGExis).toMatchSnapshot()
  })
})
