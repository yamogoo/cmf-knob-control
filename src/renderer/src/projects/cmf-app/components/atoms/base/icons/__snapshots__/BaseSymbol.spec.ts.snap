// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BaseSymbol > should render  1`] = `true`;

exports[`BaseSymbol > should render svg DOM element (path: ./symbols/AttentionSymbol.vue) 1`] = `true`;

exports[`BaseSymbol > should render svg DOM element (path: ./symbols/AutoSymbol.vue) 1`] = `true`;
